src/Arcus.Observability.Correlation/CorrelationInfoAccessorProxy.cs

```diff
@@ -0,0 +1,35 @@
+﻿using GuardNet;
+
+namespace Arcus.Observability.Correlation
+{
+    internal class CorrelationInfoAccessorProxy<TCorrelationInfo> : ICorrelationInfoAccessor
```

Comment:
Why do we need this?

Reply:
To register a `ICorrelationInfoAccessor<>` as a `ICorrelationInfoAccessor`.

Reply:
It's really about convenience.